src/publishProject/publishProjectWebViewController.ts

```diff
@@ -309,6 +323,149 @@ export class PublishProjectWebViewController extends FormWebviewController<
+     */
+    private async prepareContainerConfiguration(state: PublishDialogState): Promise<{
+        containerName: string;
+        port: number;
+    }> {
+        // Auto-generate unique container name
+        const containerName = await dockerUtils.validateContainerName("");
+
+        // Parse port (already validated by form)
+        const port = parseInt(state.formState.containerPort || constants.DefaultSqlPortNumber);
+
+        return { containerName, port };
+    }
+
+    /**
+     * Step 3: Creates Docker container using validated configuration.
+     * Runs steps 3-6: pull image, start container, check ready, connect.
+     * @param validatedContainerName - Validated unique container name
+     * @param validatedPort - Validated available port
+     * @param state Current publish dialog state
+     * @returns Connection URI if successful, error info if failed
+     */
+    private async createDockerContainer(
+        validatedContainerName: string,
+        validatedPort: number,
+        state: PublishDialogState,
+    ): Promise<{
+        success: boolean;
+        connectionUri?: string;
+        error?: string;
+        fullErrorText?: string;
+    }> {
+        // Build Docker profile using validated values
+        const dockerProfile = {
+            version: state.formState.containerImageTag || "",
+            password: state.formState.containerAdminPassword || "",
+            containerName: validatedContainerName,
+            port: validatedPort,
+            hostname: "",
+            profileName: validatedContainerName,
+            savePassword: true,
+            acceptEula: state.formState.acceptContainerLicense || false,
+        } as unknown as DockerConnectionProfile;
+
+        const dockerSteps = dockerUtils.initializeDockerSteps();
+
+        // Execute container creation steps: from pullImage to checkContainer
+        // Dynamic iteration ensures we don't miss any steps added in the future
+        for (
+            let stepIndex = DockerStepOrder.pullImage;
+            stepIndex <= DockerStepOrder.checkContainer;
+            stepIndex++
+        ) {
+            const currentStep = dockerSteps[stepIndex];
+            const args = currentStep.argNames.map(
+                (argName) => (dockerProfile as unknown as Record<string, unknown>)[argName],
+            );
+            const result = await currentStep.stepAction(...args);
+
+            if (!result.success) {
+                return {
+                    success: false,
+                    error: result.error,
+                    fullErrorText: result.fullErrorText,
+                };
+            }
+
+            // Show success message matching deployment UI format
+            void vscode.window.showInformationMessage(`✓ ${currentStep.headerText}`);
+        }
+
+        // Register connection for DacFx
+        const fileUri = `mssql://publish-container-${validatedContainerName}`;
```

Comment:
Why are we using a custom file URI here?  We should be following the same pattern as the local container dialog:

1. create the container
2. save the connection to config, establish connection, and open it in Object Explorer
3. deploy dacpac to new connection

If we're doing #2 properly, then we should have a real connection URI to use for dacFx.